### PR TITLE
Add runIf to runEach

### PR DESCRIPTION
**Pull request by DriesSchaumont**
## Describe your changes
Add `runif` to `runEach`

## Related issue(s)

Closes #xxxx 

## Type of Change

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] New functionality (non-breaking change which adds functionality)
- [ ] Major change (non-breaking change which modifies existing functionality)
- [ ] Minor change (non-breaking change which does not modify existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] This change requires a documentation update

## Checklist

Requirements:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc.
- [x] I have performed a self-review of my code by checking the "Changed Files" tab.
- [x] My code follows the code style of this project.

Tests:

- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

Documentation:

- [x] Proposed changes are described in the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/resources/io/viash/runners/nextflow/channel/runEach.nf [hunk 1]
```diff
@@ -49,7 +52,18 @@ def runEach(Map args) {
           [new_id] + tup.drop(1)
         }
         : filter_ch
-      def data_ch = id_ch | map{tup ->
+      if (runIf_) {
```
Comment (rcannood) [inline src/main/resources/io/viash/runners/nextflow/channel/runEach.nf:55]:
```suggestion
      def chRun = null
      def chPassthrough = null
      if (runIf_) {
```

**Comment by rcannood** [comment]
@DriesSchaumont Bump ;)

**Comment by DriesSchaumont** [comment]
@rcannood Note that the order in which `filter` and `runIf` are executed is the same as for the regular `.run`: first `filter` and `runIf` afterward. The fact that the `runIf` is also evaluated within the `.collect` means that the events still get multiplied times the number of input components for `.runEach` (an alternative implementation is to evaluate the `runIf` before the collect, not duplicating the events). The way it is implemented now makes it harder to just 'not touch' and event and let is pass through: it is always 'copied'. Thoughts?

**Comment by DriesSchaumont** [comment]
> > @rcannood Note that the order in which filter and runIf are executed is the same as for the regular .run: first filter and runIf afterward. Also, the fact that the runIf is also evaluated within the .collect means that the events still get multiplied times the number of input components for .runEach (an alternative implementation is to evaluate the runIf before the collect, not duplicating the events). The way it is implemented now makes it harder to just 'not touch' and event and let is pass through. Thoughts?
> 
> Interesting point! ... I _guess_ that it might not be a bad idea to swap runIf and filter around? **Breaking change alert ⚠️
> 
> At any rate, I always found it a bit dubious what the expected order of `runIf` and `filter` is. Perhaps it would have been better if filter would have had type `(id: string, state: map) -> (true | false | "remove" | "skip" | "keep")` or something like that.

WRT the breaking change: because `runIf` is not implemented yet for `runEach`, I guess its not strictly a breaking change 🤔 But the order being different for `runEach` and `run` is not ideal...

I like the idea to expand the `filter` functionality, but boy, talking about breaking changes 😅

File: src/main/resources/io/viash/runners/nextflow/channel/runEach.nf [hunk 2]
```diff
@@ -49,7 +52,20 @@ def runEach(Map args) {
...
+        id_ch.branch{ tup ->
+          run: runIf_(tup[0], tup[1], comp_)
+          passthrough: true
+        }.set{ runIfBranch }
+        chPassthrough = chPassthrough.concat(runIfBranch.passthrough)
+        chRun = chRun.concat(runIfBranch.run)
+      } else {
+        chRun = chRun.concat(id_ch)
+        chPassthrough = Channel.empty()
+      }
```
[the remark this answers is not in the document]
Comment (DriesSchaumont) [reply src/main/resources/io/viash/runners/nextflow/channel/runEach.nf]: > Is there any benefit to using `.set()` and `concat`? We can just leave the fields uninitialised until we have a value to initialise it with.

These were leftovers from when I was investigating some concurrency issues. Fixed!

**Review by rcannood — APPROVED** [review]
LGTM!

**Pull request merged**